homeassistant/components/integration/sensor.py

```diff
@@ -326,9 +325,17 @@ def calc_integration(event: Event[EventStateChangedData]) -> None:
             self.async_write_ha_state()
 
             try:
+                if old_state:
+                    old_state_state = old_state.state
+                    old_last_reported = old_state.last_reported
+                else:
+                    old_state_state = new_state.state
```

Comment:
Could you explain ? if old_state doesn't exist we simply skip the event...

Reply:
We've checked above that we either have a valid old state or or a valid old state timestamp.
If we get an old state timestamp, it means the sensor's state is unchanged.

Reply:
I see, in EventStateReportedData old_state is not required... but is it that expensive to fill it ?

Reply:
Yeah, that's not an option, because we don't create a new state object when only bumping `last_reported`: https://github.com/home-assistant/core/blob/80c8b94021bce779eacb0ff3097b29f4fea184cf/homeassistant/core.py#L2134-L2148

Reply:
but you do have old_state...

```
"old_state": old_state,
"new_state": old_state,
```

Reply:
As discussed on discord, this is not the case when only the `last_reported` timestamp is bumped